Update documentation for http probe plugin SLA support (#705)
128technology/docs: docs/plugin_http_probe.md

exit
```

Above configuration runs http probe as follow:
![Http Probe Timeline](/img/http-probe-timeline.png)

> Based on the configuration, every test runs five probe attempts. Each attempt runs with a two second hard timeout, calculated by the `probe-duration` divided by the `number-of-attempts`. Each probe may fail due to a request failure, timeout reached, or an invalid response code. Service path status is determined at the end of the probe duration and the status will remain until the end of the next test.

Reviewer: this feels like its missing a word here

> status will remain (up?/down?) until the end of the next test.

Reviewer: ```suggestion
Based on the configuration, every test runs five probe attempts. Each attempt runs with a two second hard timeout, calculated by the `probe-duration` divided by the `number-of-attempts`. Each probe may fail due to a request failure, timeout reached, or an invalid response code. Service path status is determined at the end of the probe duration and the current status will remain until the end of the next test.
```

The same steps can be used to bring `up` a path that is currently `down` by changing `STATUS=up` in the steps above.
:::

### Metrics

In-memory metrics were added for extensive http probe sla values

```
# show stats http-probe
Wed 2024-04-17 02:52:35 UTC
✔ Retrieving statistics...

Http Probe Stats Metrics
------------------------

============= ======= ==================== ====================
 Metric        Node    Probe-profile-name                Value
============= ======= ==================== ====================
 average-rtt   test2   http-probe-1                          0
 jitter        test2   http-probe-1                          0
 loss          test2   http-probe-1                        100
 max-rtt       test2   http-probe-1                          0
 min-rtt       test2   http-probe-1                          0
 updated       test2   http-probe-1         1713322347.9919806

Completed in 0.21 seconds
```

## Release Notes

### Release 1.1.0

**Release Date:** Apr 16, 2024

#### New Features and Improvements
- **PLUGIN-2300** Implement SLA monitoring per probe

> The plugin allows Users to configure the following SLA settings: max-loss, max-jitter, average-rtt. These are considered in probe success criteria. Additionally, these values are available thorugh metrics. 

Reviewer 2: ```suggestion
The plugin allows Users to configure the following SLA settings: max-loss, max-jitter, average-rtt. These settings are considered as part of the probe success criteria. Additionally, these values are available through metrics. 
```

The same steps can be used to bring `up` a path that is currently `down` by changing `STATUS=up` in the steps above.
:::

### Metrics

In-memory metrics were added for http probe SLA values

```
# show stats http-probe
Wed 2024-04-17 02:52:35 UTC
✔ Retrieving statistics...

Http Probe Stats Metrics
------------------------

===================== ======= ==================== ====================
 Metric                Node    Probe-profile-name                Value
===================== ======= ==================== ====================
 average-rtt           test2   http-probe-1                       0.12
 dns-resolution-time   test2   http-probe-1         2.1130787988658994
 jitter                test2   http-probe-1                       0.02
 loss                  test2   http-probe-1                          0
 max-rtt               test2   http-probe-1                       0.13
 min-rtt               test2   http-probe-1                       0.11
 updated               test2   http-probe-1         1718376794.8707266

Completed in 0.21 seconds
```

## Release Notes

### Release 1.2.0

> **Release Date:** Aug 19, 2024

Reviewer: ```suggestion
**Release Date:** Aug 20, 2024
```
